store/store.go

```diff
@@ -59,16 +59,31 @@ type Store interface {
 	DeleteRepo(*model.Repo) error
 
 	// GetSecretList gets a list of repository secrets
-	GetSecretList(*model.Repo) ([]*model.Secret, error)
+	GetSecretList(*model.Repo) ([]*model.RepoSecret, error)
 
 	// GetSecret gets the named repository secret.
-	GetSecret(*model.Repo, string) (*model.Secret, error)
+	GetSecret(*model.Repo, string) (*model.RepoSecret, error)
 
 	// SetSecret sets the named repository secret.
-	SetSecret(*model.Secret) error
+	SetSecret(*model.RepoSecret) error
 
 	// DeleteSecret deletes the named repository secret.
-	DeleteSecret(*model.Secret) error
+	DeleteSecret(*model.RepoSecret) error
+
+	// GetTeamSecretList gets a list of team secrets
+	GetTeamSecretList(string) ([]*model.TeamSecret, error)
+
+	// GetTeamSecret gets the named team secret.
+	GetTeamSecret(string, string) (*model.TeamSecret, error)
+
+	// SetTeamSecret sets the named team secret.
+	SetTeamSecret(*model.TeamSecret) error
+
+	// DeleteTeamSecret deletes the named team secret.
+	DeleteTeamSecret(*model.TeamSecret) error
+
+	// GetMergedSecretList gets a list of repo and team secrets
+	GetMergedSecretList(*model.Repo) ([]*model.Secret, error)
```

Comment:
we can actually merge the two lists in `store` instead of `datastore`. That way, if we have multiple implementations of datastores, we don't need to duplicate the merge logic